src/flamenco/runtime/program/fd_bpf_loader_program.c

```diff
@@ -252,7 +252,9 @@ deploy_program( fd_exec_instr_ctx_t * instr_ctx,
 
   /* Load executable */
   fd_sbpf_elf_info_t  _elf_info[ 1UL ];
-  fd_sbpf_elf_info_t * elf_info = fd_sbpf_elf_peek( _elf_info, programdata, programdata_size, deploy_mode, FD_SBPF_MAX_VERSION );
+  uint min_sbpf_version, max_sbpf_version;
+  fd_bpf_get_sbpf_versions( &min_sbpf_version, &max_sbpf_version, instr_ctx->slot_ctx );
+  fd_sbpf_elf_info_t * elf_info = fd_sbpf_elf_peek( _elf_info, programdata, programdata_size, deploy_mode, min_sbpf_version, max_sbpf_version );
```

Comment:
If a program specifies an SBPF version that hasn't been activated yet, what is the intended behaviour? From the code it looks like we treat this program as if it was an SBPF V0 program - is this intentional? Should we error out in this case, or attempt to execute it? What does Agave do?

Reply:
if you try to load an invalid version, elf_info will return as NULL, like if any other validation fails.
Tests are in `test_sbpf_elf_peek`.